app/src/components/report/MyModal.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState, useImperativeHandle, forwardRef } from 'react';
+import Box from '@mui/material/Box';
+import { Button } from 'react-bootstrap';
+import Typography from '@mui/material/Typography';
+import Modal from '@mui/material/Modal';
+
+const style = {
+    position: 'absolute' as 'absolute',
+    top: '50%',
+    left: '50%',
+    transform: 'translate(-50%, -50%)',
+    width: 400,
+    bgcolor: 'background.paper',
+    border: '2px solid #000',
+    boxShadow: 24,
+    p: 4,
+};
+
+const buttonStyle = {
+    marginTop: 10,
+};
+
+type MyModalProps = {
+    header: string;
+    description: string;
+    confirmMessage: string;
+    denyMessage: string;
+    /** Reset current location and get new location */
+    onConfirm: () => void;
+};
+
+export type ModalHandle = {
+    openModal: () => void;
+};
+
+function MyModal(
```

Comment:
Would this be more descriptively be named as `ConfirmationModal`? `MyModal` is not very descriptive :)